camera/src/test.py

```diff
@@ -0,0 +1,181 @@
+from platform import platform
+import cv2
+import tflite_runtime.interpreter as tflite
+import numpy as np
+from threading import Thread
+import os
+import time
+import subprocess
+
+CWD = os.getcwd()
+CheckpointPath = os.path.join(CWD, "model/model.tflite")
+LabelMapPath = os.path.join(CWD, "model/tflite_label_map.txt")
+
+class FFMPEGOutputStream:
+    def __init__(self, resolution=(1920, 1080), fps=30, destination_ip="192.168.1.4", destination_port="2046"):
+        command = ["ffmpeg",
+                   "-re", # Real time
+                   "-hwaccel", "auto", # Use hwaccel if you don't want to suffer
+                   "-f", "rawvideo",
+                   "-pix_fmt", "bgr24",
+                   "-s", "1920x1080",
+                   "-r", "30",
+                   "-i", "-", # Use pipe for input
+                   "-an", # No audio
+                   "-f", "mjpeg",
+                   "udp://" + destination_ip + ":" + destination_port + "?pkt_size=1316"
+                   ]
+
+        self.proc = subprocess.Popen(command, stdin=subprocess.PIPE)
+
+    def __del__(self):
+        self.proc.stdin.close()
+        self.proc.terminate()
+
+    def submit(self, frame, labels):
+        timer_start = time.perf_counter()
+        for result in labels:
+            (lower_left, upper_right) = result["box"]
+            (box_x_min, box_y_min) = lower_left
+            name = result["name"]
+            score = result["score"]
+
+            cv2.rectangle(frame, lower_left, upper_right, (10, 255, 0), 2)
+
+            # Draw label
+            label = '%s: %d%%' % (name, score)
+            labelSize, baseLine = cv2.getTextSize(label, cv2.FONT_HERSHEY_SIMPLEX, 0.7, 2) # Get font size
+            label_ymin = max(box_y_min, labelSize[1] + 10)
+            cv2.rectangle(frame, (box_x_min, label_ymin-labelSize[1]-10), (box_x_min+labelSize[0], label_ymin+baseLine-10), (255, 255, 255), cv2.FILLED) # Draw white box to put label text in
+            cv2.putText(frame, label, (box_x_min, label_ymin-7), cv2.FONT_HERSHEY_SIMPLEX, 0.7, (0, 0, 0), 2) # Draw label text
+        
+        print("submit: " + str(time.perf_counter() - timer_start))
+        self.proc.stdin.write(frame.tobytes())
+
+class WebcamStream:
+    udpStreamEnabled = False
+    running = True
+    frame = None
+
+    def __init__(self, resolution=(1920, 1080), fps=30):
+        camera = cv2.VideoCapture(0)
+        if not camera.isOpened():
+            print("Error opening camera")
+            exit(-1)
```

Comment:
Does this exit out of the whole program?  If there's an error here, should we be exiting out of the software?

Reply:
Likely be running it as a systemd service or smth to automatically restart. Way easier to do that then to handle restarting the camera over and over and over when it may not even be present.